test/test-proxied-properties.js

```diff
@@ -162,6 +163,16 @@ describe("browser-polyfill", () => {
         get tabs() {
           ok(false, "chrome.tabs should not lazily be initialized without explicit API call");
         },
+        privacy: {
+          get network() {
+            ++lazyInitPrivacyCount;
+
+            const networkPredictionEnabled = () => true;
+            const value = {networkPredictionEnabled};
+            Object.defineProperty(fakeChrome.privacy, "network", {value});
```

Comment:
Don't replace the property. Now we're interested in how often `privacy.network` is dereferenced.

Reply:
Noted, done.

It fails, but I can confirm it's not a regression as it was calling twice for the old implementation as well.

Any idea why it's so? Might be a problem with my test?

Reply:
Seems like if I replace getter in [the implementation](https://github.com/mozilla/webextension-polyfill/pull/205/files#diff-a8dc0200482fd536ca439ed8bcf763c2R505) the object will be dereferenced once.

Will do so if no objection.

Reply:
Nevermind, it doesn't help.

Reply:
> It fails, but I can confirm it's not a regression as it was calling twice for the old implementation as well.

This was wrong assumption, apparently I was running test without running `npm run build` first, Sorry for this, will have a look into it later today, the old tests were passing correctly, the problem is in the new implementation.

Reply:
After taking a closer look into implementation, now I see what is the problem.

The problem is that I can't fix it without touching `wrapAPIs()` code, unless I hardcode everything.

I'm thinking about something like:

```js
const settingMetadata = {
      clear: {minArgs: 1, maxArgs: 1},
      get: {minArgs: 1, maxArgs: 1},
      set: {minArgs: 1, maxArgs: 1},
};

apiMetadata.privacy = {"*": {"*": settingMetadata}};
```

and updating `wrapObject()`:

```
if (hasOwnProperty(metadata, "*")) {
 value = wrapObject(value, wrappers[prop], metadata["*"]);
}
```

I'm quite uncomfortable updating `wrapObject()`, but I'll give it a shot and revert it if you say so.

Reply:
@Rob--W Done.